src/main.tsx

```diff
@@ -34,7 +31,12 @@ const router = createBrowserRouter([
 			{
 				path: 'dashboard',
 				element: <DashboardScreen />
-			}
+			},
+			{
+				path: 'search',
+				element: <SearchResultsScreen />,
+				loader: searchLoader
+			} 
```

Comment:
`loader` function will return the query as a string, then attach it to the URL `/search/loader-string`?

Reply:
> `loader` function will return the query as a string, then attach it to the URL `/search/loader-string`?

No, loader will return the string to the loaderData in the search route. Then it is accessed inside the component with `useLoaderData`. The loader is actually pulling the query from the URL.